architecture-decisions/0003-preservation.md

```diff
@@ -0,0 +1,84 @@
+# 3. Preservation
+
+Date: 2019-04-02
+
+## Status
+
+Accepted
+
+## Context
+
+We have agreed that we will preserve digital objects by saving resources in
+Google Cloud Storage in a directory structure which preserves both the binaries
+the resource is made up of as well as the JSON serialization of the resource
+itself.
+
+## Decisions
+
+1. Preserving
+   1. We will preserve materials in Google Cloud Coldline Storage with
+      `versioning` enabled. Versions will be kept indefinitely and without
+      limit. All files will go in a single bucket.
+   1. When a resource is `complete` and marked with the `cloud` preservation
+      policy it will save itself and all resources contained in `member_ids` in
+      a directory structure in Google Cloud Storage that looks like the
+      following:
+      ```
+      - <resource-id>
+        - data
+          - <child-id>
+            - <child-id>.json
+            - <binary.tif>
+        - <resource-id>.json
+      ```
+   1. When a child with a preserved parent (any length up the hierarchy) is
+      `complete` and marked with the `cloud` preservation policy and is updated,
+      it will update itself.
+   1. Related objects such as collections, Ephemera Terms, etc. will not be
+      packaged inside the preserved object. If it's important they be preserved,
+      those objects should be marked with the `cloud` preservation policy.
+   1. When a FileSet is added to a resource which is already complete and marked
+      with the `cloud` preservation policy, it will upload the new binary
+      content to both the repository and to Google Cloud STorage
+   1. If a child is marked to be preserved, but its parent is not, it will still
+      save in a nested directory structure, but will not automatically create
+      backups of its parents.
+   1. This behavior will be attached to the ChangeSetPersister.
+1. Packaging Details
+   1. Resources created in cloud storage will be attached to the appropriate
+      resource in `file_metadata` with a PCDM Use value of
+      `pcdm:PreservationMasterFile`
```

Comment:
We're currently using `pcdm:PreservationMasterFile` for audio master files.  Do we want to avoid using the same value, or is the different storage adapter enough to keep them separate?

Reply:
Yeah, when I actually implemented this I ended up using PreservationCopy and PreservedMetadata. Maybe we should block this and see how we like that?

Reply:
What about "PreservationFile" and "PreservationMetadata"? "Copy" feels potentially vague.